src/Algebra/Graph/Relation/Symmetric/Internal.hs

```diff
@@ -0,0 +1,193 @@
+-----------------------------------------------------------------------------
+-- |
+-- Module     : Algebra.Graph.Relation.Symmetric.Internal
+-- Copyright  : (c) Andrey Mokhov 2016-2019
+-- License    : MIT (see the file LICENSE)
+-- Maintainer : andrey.mokhov@gmail.com
+-- Stability  : unstable
+--
+-- This module exposes the implementation of derived binary relation data types.
+-- The API is unstable and unsafe, and is exposed only for documentation. You
+-- should use the non-internal module "Algebra.Graph.Relation.Symmetric" instead.
+-----------------------------------------------------------------------------
+
+module Algebra.Graph.Relation.Symmetric.Internal (
+    -- * Implementation of derived binary relations
+    SymmetricRelation (..), empty, vertex, overlay, connect, consistent,
+    referredToVertexSet, deduplicateSet, deduplicate
+  ) where
+
+import Control.DeepSeq (NFData (..))
+
+import Algebra.Graph.Relation.Internal hiding (empty, vertex, overlay, connect, referredToVertexSet, consistent)
+import Algebra.Graph.Relation (domain, relation, transpose)
+import qualified Data.Set as Set
+import Data.Monoid (mconcat)
+
+{-|  The 'SymmetricRelation' data type represents a /symmetric binary relation/
+over a set of elements. Symmetric relations satisfy all laws of the
+'Undirected' type class and, in addition, the
+commutativity of connect:
+
+@'connect' x y == 'connect' y x@
+
+The 'Show' instance produces symmetrically closed expressions:
```

Comment:
In which sense are these results "symmetrically closed"? I think this is probably an outdated comment and you should explain here that the vertices of an edge are listed in the non-decreasing order.

Reply:
It's worth giving an example, which clarifies this, e.g. adding `2 * 1` as well.

Reply:
I also added the example of `1 * 2 * 3` and `3 * 2 * 1` to clarify the non-decreasing order.

Reply:
Perhaps, `3*2*1` and `1*2*1` would be a bit more informative.

Reply:
I will change to that!